sched/task/task_setcanceltype.c

```diff
@@ -100,7 +100,14 @@ int task_setcanceltype(int type, FAR int *oldtype)
 #ifndef CONFIG_DISABLE_PTHREAD
           if ((tcb->flags & TCB_FLAG_TTYPE_MASK) == TCB_FLAG_TTYPE_PTHREAD)
             {
+              tcb->flags &= ~TCB_FLAG_CANCEL_PENDING;
+              tcb->flags |= TCB_FLAG_CANCEL_DOING;
+#if !defined(CONFIG_BUILD_FLAT) && defined(__KERNEL__)
```

Comment:
why not move TCB_FLAG_CANCEL_DOING/TCB_FLAG_CANCEL_PENDING to tls_info_s? so task_setcanceltype can move libc.

Reply:
> why not move TCB_FLAG_CANCEL_DOING/TCB_FLAG_CANCEL_PENDING to tls_info_s? so task_setcanceltype can move libc.

Good idea.  But isn't that a different PR?